src/FileInput/FileInput.tsx

```diff
@@ -204,6 +204,10 @@ export class FileInput extends React.Component<Props, State> {
                 "FileInput",
                 formElementSizeClassName(this.props.size),
                 this.props.className,
+                !!this.state.filename &&
+                  !this.state.success &&
+                  !this.state.error &&
+                  "FileInput--AttachmentIsUploading",
```

Comment:
It seems like the styles are only getting applied from the MessagingInput `.less` file but we would still end up adding an extra class for every usage of `FileInput`, which might not be ideal. 

Was thinking it might make sense to add this class to the top level component (MessagingInput) and apply it from there. Something like

```
.MessagingInput--AttachmentIsUpload .FileInput { ... }
```

Reply:
I thought about this too. The issue is that the class is added here based on the state of the `FileInput`. I'm not sure exactly how to pass that info up to the `MessagingInput`, so the class could be set there. Open to ideas tho!

Reply:
Ah yeah, that's a good point. The current approach makes sense then!